lib/utils/logger.js

```diff
@@ -90,42 +90,123 @@ const methods = {
 
 Object.freeze(methods);
 
+function timestamp() {
+	return moment().format('MMM D HH:MM:SS');
+}
+
 /**
- * Logger function.
- * @param {*} method
+ * Prefix log with device id
+ * @param {object} deviceDetails
+ */
+function devicePrefix(deviceDetails) {
+	if (!deviceDetails || !deviceDetails.shortDisplayName)
+		return 'Launcher ';
+
+	return deviceDetails.shortDisplayName + ' ';
+}
+
+function logLine(msg, logMethod, fontColor, leftRail = '') {
+	let obj = msg;
+
+	if (typeof msg === 'string') {
+		try {
+			obj = JSON.parse(msg);
+		} catch (e) {
+			// something to make ESLINT happy
+		}
+	}
+
+	const lines = (typeof obj === 'object') ?
+		jsonRenderer.render(obj) : fontColor(String(msg));
+
+	lines.split('\n').forEach(line => {
+		logMethod(leftRail + line);
+	});
+
+	if (typeof obj === 'object')
+		logMethod(leftRail);
+}
+
+function extractOpType(messages) {
+	// first item could be the operation type (assert or evaluate)
+	let logMessages = messages;
+	const {opType, logString} = formatOpType(messages[0]);
+
+	if (opType) {
+		logMessages = logMessages.slice(1);
+		if (logString)
+			logMessages = [logString, ...logMessages];
+	}
+
+	return {
+		opType,
+		logMessages,
+	};
+}
+
+let lastLogSource = '';
+
+/**
+ *
+ * @param consoleMethod
+ * @param messages (last argument should be the device
+ * @param logMethod
+ * @returns {*[]}
  */
-const log = method => (message, deviceId) => {
-	const currentTimestamp = config.timestamp.toLowerCase() === timestampNone
-		? ''
-		: `[${timestamp(config.timestamp)}] `;
+function outputLogsInLauncherMode(consoleMethod, messages, logMethod) {
+	const fontColor = consoleMethod.launcherColor || stColors.suit;
+	const deviceDetails = messages[messages.length - 1];
```

Comment:
I think `const log = method => (message, deviceId) => {` is much clearer interface

Reply:
I think so too, however this is how it is implemented in master currently, I only refactored this a bit due to CodeClimate complains. 
The problem is that `logger` is used for both the launcher and the child process. And when the Launcher captures logs from the child process this is what the logger gets: the captured message and device information from the process.